addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_image.py

```diff
@@ -149,64 +150,26 @@ def split_pixels_by_channels(image: bpy.types.Image, export_settings) -> typing.
                         'B': 2
                     }[elem.from_socket.name]
 
-            if source_channel is not None:
-                pixels = [split_pixels_by_channels(result.shader_node.image, export_settings)[source_channel]]
-                target_channel = source_channel
-                source_channel = 0
-                source_channels_length = 1
-            else:
-                pixels = split_pixels_by_channels(result.shader_node.image, export_settings)
-                target_channel = 0
-                source_channel = 0
-                source_channels_length = len(pixels)
+            image = gltf2_blender_image.ExportImage.from_blender_image(result.shader_node.image)
+
+            if composed_image is None:
+                composed_image = gltf2_blender_image.ExportImage.zero_image(image.width, image.height)
 
             # Change target channel for metallic and roughness.
             if elem.to_socket.name == 'Metallic':
```

Comment:
This line, and the one 2 below it, didn't change, but I think they have been harboring a bug.  `elem.to_socket.name` is the name of the output of the farthest-left link, the last one traversed by the `for` loop above.  I'm pretty sure these lines should both change to just `if socket.name ==`, which will look at the final destination of the links.